PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/helpers/UIHelperImpl.java

```diff
@@ -645,6 +696,47 @@ public static ImageIcon loadIcon(@Nullable String name) {
         return new ImageIcon(url);
     }
 
+    @Override
+    public Icon loadIconByAction(AzureActionEnum actionEnum) {
+        if (AzureActionEnum.REFRESH.equals(actionEnum)) {
+            return AzureAllIcons.Common.RESTART;
+        } else if (AzureActionEnum.CREATE.equals(actionEnum)) {
+            return AzureAllIcons.Common.CREATE;
+        } else if (AzureActionEnum.START.equals(actionEnum)) {
+            return AzureAllIcons.Common.START;
+        } else if (AzureActionEnum.STOP.equals(actionEnum)) {
+            return AzureAllIcons.Common.STOP;
+        } else if (AzureActionEnum.RESTART.equals(actionEnum)) {
+            return AzureAllIcons.Common.RESTART;
+        } else if (AzureActionEnum.DELETE.equals(actionEnum)) {
+            return AzureAllIcons.Common.DELETE;
+        } else if (AzureActionEnum.OPEN_IN_PORTAL.equals(actionEnum)) {
+            return AzureAllIcons.Common.OPEN_IN_PORTAL;
+        } else if (AzureActionEnum.SHOW_PROPERTIES.equals(actionEnum)) {
+            return AzureAllIcons.Common.SHOW_PROPERTIES;
+        } else if (AzureActionEnum.START.equals(actionEnum)) {
+            return AzureAllIcons.Common.START;
+        }
+        return null;
+    }
+
+    @Override
+    public Icon loadIconByNodeClassWithStates(Class<? extends Node> clazz, boolean running, boolean updating) {
```

Comment:
We may use Enum like `Status` and using `Map<Status, Icon>` as the value type of `NodeIconsMap`

Reply:
By the way, shall we use enum for common icons in utils and add `loadIconByEnum(IconEnum)` in `UIHelper`?